GithubQuickApprove.user.js

```diff
@@ -78,44 +76,81 @@ const insertButton = () => {
-        console.log({ authenticity_token, head_sha, csrf })
-        console.log('Missing data, cannot create Quick Approve button')
-        return
+      const parser = new DOMParser();
+      const responseXML = parser.parseFromString(xhr.responseText, "text/html");
+
+      const reviewForm = responseXML.querySelector(
+        "form#pull_requests_submit_review"
+      );
+
+      const csrfInputValue = reviewForm.querySelector(
+        'input[data-csrf="true"]'
+      ).value;
+      const authenticityTokenInputValue = reviewForm.querySelector(
+        'input[name="authenticity_token"]'
+      ).value;
+      const headShaInputValue = reviewForm.querySelector(
+        'input[name="head_sha"]'
+      ).value;
+
+      if (
+        !csrfInputValue ||
+        !authenticityTokenInputValue ||
+        !headShaInputValue
+      ) {
+        console.error("Error: Could not find required input values");
+        return;
       }
-      authenticityTokenInput.setAttribute('value', authenticity_token)
-      headShaInput.setAttribute('value', head_sha)
-      csrfInput.setAttribute('value', csrf)
+
+      csrfInput.setAttribute("value", csrfInputValue);
+      authenticityTokenInput.setAttribute("value", authenticityTokenInputValue);
+      headShaInput.setAttribute("value", headShaInputValue);
 
       const headerActions =
-        document.getElementsByClassName('gh-header-actions')[0]
-      headerActions.append(form)
+        document.getElementsByClassName("gh-header-actions")[0];
+      headerActions.append(form);
     }
+  };
+  xhr.open("GET", `${githubHost}/${paths.join("/")}/files`);
+  xhr.send();
+
+  const headerActions = document.querySelector(".gh-header-actions");
+  if (headerActions) {
+    headerActions.append(form);
   }
-  xhr.open('GET', 'https://github.com/' + paths.join('/') + '/files')
-  xhr.send()
-}
-
-const observeUrlChange = () => {
-  let oldHref = document.location.href
-  const body = document.body
-  const observer = new MutationObserver((mutations) => {
-    if (oldHref !== document.location.href) {
-      oldHref = document.location.href
-      if (document.location.pathname.match(/\/pull\/\d+/)) insertButton()
+};
+
+let oldHref = document.location.href;
+const githubHost = window.location.origin;
+const documentMutationObserver = new MutationObserver((_) => {
+  if (oldHref !== document.location.href) {
+    oldHref = document.location.href;
+    if (document.location.pathname.match(/\/pull\/\d+/)) {
+      insertButton();
     }
-  })
-  observer.observe(body, { childList: true, subtree: true })
-  insertButton()
-}
+  }
+});
 
-window.onload = observeUrlChange
+document.addEventListener("DOMContentLoaded", () => {
+  documentMutationObserver.observe(document.body, {
+    childList: true,
+    subtree: true,
+  });
+  const path = window.location.pathname;
+  if (path.match(/\/pull\/\d+/)) {
+    insertButton();
+  }
+});
+window.onload = documentMutationObserver;
```

Comment:
Since `documentMutationObserver` is an instance of `MutationObserver`, this event handler won't be executed.